### PR TITLE
parser: fix alter_stmt order by syntax

### PR DESCRIPTION
**Pull request by zuoRambo**
### What problem does this PR solve? 
Fix ORDER BY syntax. Issue: https://github.com/pingcap/parser/issues/499

### MySQL Syntax:
```
alter_specification:
...
  | ORDER BY col_name [, col_name] ...
```

### Bad SQL Case:

```sql
ALTER TABLE t_n ORDER BY ident
```

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported variable/fields change
 - Has interface methods change

Related changes

 - Need to update the documentation

### CODE REVIEW

File: ast/ddl.go [hunk 1]
```diff
@@ -2161,6 +2165,16 @@ func (n *AlterTableSpec) Restore(ctx *RestoreCtx) error {
...
 		ctx.WritePlain("= ")
 		ctx.WriteKeyWord(n.LockType.String())
+	case AlterTableOrderByColumns:
+		ctx.WriteKeyWord("ORDER BY ")
+		for i, columnName := range n.OrderByColumns {
+			if i != 0 {
+				ctx.WritePlain(",")
+			}
+			if err := columnName.Restore(ctx); err != nil {
+				return errors.Annotate(err, fmt.Sprintf("An error occurred while restore AlterTableSpec.OrderByColumns[%d]", i))
```
Comment (leoppro) [inline ast/ddl.go]:
```suggestion
				return errors.Annotatef(err, "An error occurred while restore AlterTableSpec.OrderByColumns[%d]", i)
```

File: parser.y [hunk 2]
```diff
@@ -213,7 +213,7 @@ import (
 	realType		"REAL"
 	references		"REFERENCES"
 	regexpKwd		"REGEXP"
-	rename         		"RENAME"
+	rename          "RENAME"
```
Comment (leoppro) [inline parser.y]:
```suggestion
	rename         		"RENAME"
```

File: parser.y [hunk 3]
```diff
@@ -1156,6 +1156,8 @@ import (
 %precedence local
 %precedence lowerThanRemove
 %precedence remove
+%precedence lowerThenOrderBy
+%precedence orderBy
```
Comment (leoppro) [inline parser.y]:
```suggestion
%precedence order
```

File: parser.y [hunk 4]
```diff
@@ -1156,6 +1156,8 @@ import (
 %precedence local
 %precedence lowerThanRemove
 %precedence remove
+%precedence lowerThenOrderBy
```
Comment (leoppro) [inline parser.y]:
```suggestion
%precedence lowerThenOrder
```

File: parser.y [hunk 5]
```diff
@@ -1510,6 +1512,13 @@ AlterTableSpec:
 			Name: $5.(string),
 		}
 	}
+|	"ORDER" "BY" ColumnNameList %prec lowerThenOrderBy
```
Comment (leoppro) [inline parser.y]:
```suggestion
|	"ORDER" "BY" ColumnNameList %prec lowerThenOrder
```

File: parser.y [hunk 6]
```diff
@@ -4534,7 +4543,7 @@ StringLiteral:
 
 
 OrderBy:
-	"ORDER" "BY" ByList
+	"ORDER" "BY" ByList %prec orderBy
```
Comment (leoppro) [inline parser.y]:
```suggestion
	"ORDER" "BY" ByList
```

**Comment by leoppro** [comment]
@tangenta PTAL

File: parser.y [hunk 7]
```diff
@@ -1510,6 +1514,13 @@ AlterTableSpec:
 			Name: $5.(string),
 		}
 	}
+|	"ORDER" "BY" AlterOrderList %prec lowerThenOrder
+	{
+		$$ = &ast.AlterTableSpec{
+            Tp: ast.AlterTableOrderByColumns,
+            OrderByList: $3.([]*ast.AlterOrderItem),
```
Comment (tangenta) [inline parser.y]: Ditto.

File: parser.y [hunk 8]
```diff
@@ -4540,6 +4551,21 @@ StringLiteral:
...
+|   AlterOrderList ',' AlterOrderItem
+    {
+        $$ = append($1.([]*ast.AlterOrderItem), $3.(*ast.AlterOrderItem))
+    }
+
+AlterOrderItem:
+    ColumnName Order
+    {
+        $$ = &ast.AlterOrderItem{Column: $1.(*ast.ColumnName), Desc: $2.(bool)}
+    }
```
Comment (tangenta) [inline parser.y]: Ditto.

File: parser.y [hunk 9]
```diff
@@ -878,6 +878,8 @@ import (
 	ByItem				"BY item"
 	OrderByOptional			"Optional ORDER BY clause optional"
 	ByList				"BY list"
+    AlterOrderItem  "Alter Order item"
+    AlterOrderList  "Alter Order list"
```
Comment (tangenta) [inline parser.y]: Please indent these lines with tab.

**Pull request merged**